marathon/models/deployment.py

```diff
@@ -29,14 +30,19 @@ def __init__(
         ]
         self.current_step = current_step
         self.id = id
-        self.steps = [
-            [step if isinstance(step, MarathonDeploymentAction) else MarathonDeploymentAction().from_json(step)
-                for step in s]
-            for s in (steps or [])
-        ]
+        self.steps = [self.parse_deployment_step(step) for step in (steps or [])]
         self.total_steps = total_steps
         self.version = version
 
+    def parse_deployment_step(self, step):
+        if step.__class__ == dict:
+            # This is what Marathon 1.0.0 returns: steps
+            return MarathonDeploymentStep().from_json(step)
+        elif step.__class__ == list:
+            # This is Marathon < 1.0.0 style, a list of actions
+            return [s if isinstance(s, MarathonDeploymentAction) else MarathonDeploymentAction().from_json(s) for s in step]
```

Comment:
it's a bit of a code smell that steps could be an array of MarathonDeploymentAction or an array of strings ?(or both?). I know you're just moving code around, but I'd be interested in tracking down what use this is covering